services/FacilityService.php

```diff
@@ -0,0 +1,246 @@
+<?php
+/**
+ * FacilityService
+ *
+ * Copyright (C) 2017 Matthew Vita <matthewvita48@gmail.com>
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * @author  Matthew Vita <matthewvita48@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+
+namespace services;
+
+class FacilityService
+{
+    /**
+     * Default constructor.
+     */
+    public function __construct()
+    {
+    }
+
+    public function getAll()
+    {
+        return $this->get(array("order" => "ORDER BY FAC.name ASC"));
+    }
+
+    public function getPrimaryBusinessEntity($options = null)
+    {
+        if (!empty($options) && !empty($options["useLegacyImplementation"])) {
+            return getPrimaryBusinessEntityLegacy();
+        }
+
+        $args = array(
+            "where" => "WHERE FAC.primary_business_entity = 1",
+            "data"  => null,
+            "limit" => 1
+        );
+
+        if (!empty($options) && !empty($options["excludedId"])) {
+            $args["where"] .= " AND FAC.id != ?";
+            $args["data"] = $options["excludedId"];
+            return $this->get($args);
+        }
+
+        return $this->get($args);
+    }
+
+    public function getAllServiceLocations($options = null)
+    {
+        $args = array(
+            "where" => null,
+            "order" => "ORDER BY FAC.name ASC"
+        );
+
+        if (!empty($options) && !empty($options["orderField"])) {
+            $args["order"] = "ORDER BY FAC." . escape_sql_column_name($options["orderField"]) . " ASC";
```

Comment:
note this function is much more inefficient if you give the table in the second parameter. if give table then it will just collect sql column names from the chosen table and ensures it exists (ie. uses whitelisting). however, if don't give a table, then it collect every sql column name from the entire database... :)